include/revng/ABI/Definition.h

```diff
@@ -147,6 +147,13 @@ type: struct
       Setting this to true disables usage of vector registers entirely.
     type: bool
 
+  - name: AllowPuttingSimilarlyAlignedStackArgumentsTogether
+    doc: |
+      Setting this to true allows stack arguments to use struct-like alignment
+      rules instead of the common word-extension system. A notable ABI using
```

Comment:
I've no idea what the "word-extension system" is.
Please make examples on different behavior. Without examples, all of this documentation is rather obscure.

Reply:
In practically all the ABIs, no argument can be smaller than a word, as in, they always get extended to a word (no matter the size, be it 1 byte or 2) before doing any allocation. Setting this to `true`, disables the behavior - making them more struct-like.

Reply:
OK, add what you said in the doc.

Reply:
Done.

```
Setting this to true allows stack arguments to use slots smaller than a word (makes them behave like a single struct instead of a bunch of arguments): so they don't get extended unnecessarily. A notable ABI using this is apple flavor of the aarch64.

For example

void function(/* enough arguments of all the registers */, uint8_t, uint8_t);

When this is set to true, both `uint8_t` will get placed as if they were a struct, that is without any padding in-between. When this is false, both of them will get extended to register size before getting placed, leading to padding.
```